src/main/java/org/codeforamerica/messaging/services/MessageService.java

```diff
@@ -20,13 +20,22 @@ public class MessageService {
     private final SmsService smsService;
     private final EmailService emailService;
     private final MessageRepository messageRepository;
+    private final TemplateSetRepository templateSetRepository;
+    private final TemplateVariantRepository templateVariantRepository;
```

Comment:
I didn't think that we would need a `TemplateVariantRepository`. Wouldn't it make more sense to use `FetchType.EAGER` on the `OneToMany`?

Reply:
Thanks for pointing that out, I'll change it

Reply:
This was surprisingly hard to figure out, though ultimately it came down to getting the JPA annotations right, and I'm not sure I nailed it. The test data setup in particular could probably use some work.